tests/cli/lib/root_account.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+check_root_account() {
+# Try to SSH to a remote machine first using root account using password-based
+# auth (this is expected to fail) and then using key-based auth with the
+# supplied username to check content of /etc/shadow and audit.log.
+#
+# use: check_root_account <user> <machine> [ssh options]
+
+    local ssh_opts="-o StrictHostKeyChecking=no $3"
+    local user="$1"
+    local machine="$2"
+    if [[ "$user" == "" || "$machine" == "" ]]; then
+        rlFail "check_root_account: Missing user or machine parameter."
+        return 1
+    fi
+
+    # ssh returns 255 in case of any ssh error, so it's better to grep the specific error message
+    rlRun -t -c "ssh $ssh_opts -o PubkeyAuthentication=no root@${machine} 2>&1 | grep -i 'permission denied ('" \
+        0 "Can't ssh to '$machine' as root using password-based auth"
+    rlRun -t -c "ssh $ssh_opts ${user}@${machine} \"sudo grep -E '^root:(\*LOCK\*|!):' /etc/shadow\"" \
+        0 "root account is disabled in /etc/shadow"
+    rlRun -t -c "ssh $ssh_opts ${user}@${machine} \"sudo grep 'USER_LOGIN.*acct=\\\"root\\\".*terminal=ssh.*res=failed' /var/log/audit/audit.log\"" \
+        0 "audit.log contains entry about unsuccessful root login"
+    rlRun -t -c "ssh $ssh_opts ${user}@${machine} 'cat /etc/redhat-release'"
+}
+
+check_root_account_live_iso() {
+# Inspect state of root account in live ISO instance
+#
+# use: check_root_account_live_iso <user> <machine> [ssh options]
+
+    local ssh_opts="-o StrictHostKeyChecking=no -o PubkeyAuthentication=no $3"
```

Comment:
`-o PubkeyAuthentication=no` is going to fail for live-iso b/c it can only support key based authentication